src/bartender/web/api/job/views.py

```diff
@@ -274,3 +279,95 @@ async def retrieve_job_directories_from_path(
         ) from exc
     current_depth = len(start_dir.relative_to(job_dir).parts)
     return await walk_dir(start_dir, job_dir, current_depth + max_depth)
+
+
+def _remove_archive(filename: str) -> None:
+    """Remove archive after file response.
+
+    Args:
+        filename: path to the file that should be removed.
+    """
+    Path(filename).unlink()
+
+
+@router.get(
+    "/{jobid}/archive",
+    responses={200: {"content": {"application/octet-stream": {}}}},
+)
+async def retrieve_job_directory_as_archive(
+    job_dir: CurrentCompletedJobDir,
+    background_tasks: BackgroundTasks,
+    archive_fmt: str = ".zip",
+    exclude: Optional[list[str]] = Query(default=None),
+    exclude_dirs: Optional[list[str]] = Query(default=None),
+    # Note: also tried to with include (filter & filter_dirs) but that can be
+    # unintuitive. e.g. include_dirs=['output'] doesn't return subdirs of
+    # /output that are not also called output. Might improve when globs will be
+    # supported in next release (already documented):
+    # https://github.com/PyFilesystem/pyfilesystem2/pull/464
+) -> FileResponse:
+    """Download contents of job directory as archive.
+
+    Args:
+        job_dir: The job directory.
+        background_tasks: FastAPI mechanism for post-processing tasks
+        archive_fmt: Format to use for archive. Supported formats are '.zip', '.tar',
+            '.tar.xz', '.tar.gz', '.tar.bz2'
+        exclude: list of filename patterns that should be excluded from archive.
+        exclude_dirs: list of directory patterns that should be excluded from archive.
+
+    Returns:
+        FileResponse: Archive containing the content of job_dir
+
+    """
+    dst_fs = _parse_archive_fmt(archive_fmt)
+    archive_fn = str(job_dir.with_suffix(archive_fmt))
+    with (  # noqa: WPS316
+        OSFS(str(job_dir)) as src,
+        dst_fs(archive_fn, write=True) as dst,
+    ):
+        copy_fs(src, dst, walker=Walker(exclude=exclude, exclude_dirs=exclude_dirs))
```

Comment:
Given dir to archive is big like several Gigabytes and while copy_fs() is running  then I think no other requests can be handled in the current process.

Could you make it async with

```suggestion
        async_copy_fs = async_wrap(copy_fs)
        await async_copy_fs(src, dst, walker=Walker(exclude=exclude, exclude_dirs=exclude_dirs))
```

Then it will run in a thread pool, freeing up the event loop.